### DISCUSSION TITLE
Сервис ломает upnp

### DISCUSSION THREAD

**Discussion by egorovna-developments** [xtrime-ru/antizapret-vpn-docker#47] [Q&A] [answered]
Соединение настроено на роутере openwrt. UPNP сервис использует stun сервер google для определения внешнего ip. Есть подозрение что vpn его зачем-то резолвит. 

stun.l.google.com:19302

Добавил stun.l.google.com в exclude-hosts-custom.txt. Не помогло. Без stun сервера не поднимается upnp.

**Comment by Selys-u** [comment]
а nslookup что говорит?

**Comment by egorovna-developments** [comment]
Server:		127.0.0.1
Address:	127.0.0.1:53

Non-authoritative answer:
Name:	stun.l.google.com
Address: 74.125.250.129

Non-authoritative answer:
Name:	stun.l.google.com
Address: 2001:4860:4864:5:8000::1

**Reply by Selys-u** [reply]
Сделайте трассировку

**Reply by egorovna-developments** [reply]
traceroute to stun.l.google.com (74.125.250.129), 20 hops max, 46 byte packets
 1  5.189.246.1  6.958 ms
 2  172.29.129.38  0.331 ms
 3  5.189.241.114  0.673 ms
 4  188.170.166.22  4.150 ms
 5  10.222.24.13  56.527 ms
 6  209.85.168.242  60.319 ms
 7  *
 8  108.170.236.34  55.725 ms
 9  192.178.241.70  53.718 ms
10  192.178.240.239  68.938 ms
11  142.250.235.74  65.530 ms
12  142.250.209.171  75.021 ms
13  *
14  *
15  *
16  *
17  *
18  *
19  *
20  *

**Comment by xtrime-ru** [comment]
> Добавил stun.l.google.com в exclude-hosts-custom.txt. Не помогло. Без stun сервера не поднимается upnp.

Резолв всех доменов идет через DNS на сервере.  Возможно причина в том, что exclude-hosts-custom.txt не рассчитан на домены выше второго уровня. Попробуйте exclude-regex-custom.txt:  https://github.com/xtrime-ru/antizapret-vpn-docker?tab=readme-ov-file#adding-domainsips

**Reply by egorovna-developments** [reply]
\\.google\\.com$ так?

**Reply by xtrime-ru** [reply]
Ваш вариант все поддомены гугла исключит. Будет работать. 
Если надо конкретный, то так: 
```
stun\.l\.google\.com$
```

**Reply by egorovna-developments** [reply]
Fri Aug 23 15:29:03 2024 user.notice upnp daemon: external interface not found, not starting
Fri Aug 23 15:29:04 2024 daemon.err miniupnpd[2163]: STUN: Cannot get IP address for ext interface
Fri Aug 23 15:29:04 2024 daemon.err miniupnpd[2163]: Performing STUN failed. EXITING

без изменений

**Reply by xtrime-ru** [reply]
Надо гуглить ошибку `upnp daemon: external interface not found, not starting`
Проблема вообще не в openvpn. Вероятно надо как то дополнительно настраивать роутер, что бы внутренние сервисы не конфликтовали с openvpn

**Reply by egorovna-developments** [reply]
Да, похоже проблема пришла с каким-то обновлением прошивки. Потому что upnp ломает даже просто поднятое соединение openvpn. Даже без всех переадресаций на него.

**Reply by egorovna-developments** [reply]
Постепенно наращивая настройку нашёл что ломают UPNP параметры в конфиге
route 77.88.8.8
route 8.8.8.8
Я их добавлял для работы оригинального антизапрет контейнера. Из-за того что в нём был DNS от гугл. А на вашем нужно? Без этого вроде не работает.

**Reply by xtrime-ru** [reply]
У нас используется системный DNS, 8.8.8.8, yandex DNS для доменов apple, и государственный DNS для ру, рф и su доменов. 
 Если нужен переопредить то достаточно указать в ENV параметрах DNS=8.8.8.8
 
На некоторых системах внутри контейнера не работает DNS, тогда нужно раскоментировать строки: https://github.com/xtrime-ru/antizapret-vpn-docker/blob/master/docker-compose.yml#L21-L22
 и сделать `docker compose down && docker compose up -d`

**Reply by egorovna-developments** [reply]
Похоже дело всё же в конфиге. Потому что настройка от antizapret работает с upnp.

**Reply by egorovna-developments** [reply]
nobind
client

remote v.31337.lol

route 77.88.8.8
route 8.8.8.8

remote-cert-tls server

dev tun
proto tcp
cipher AES-128-CBC
setenv opt data-ciphers AES-128-GCM:AES-256-GCM:AES-128-CBC

resolv-retry infinite
persist-key
persist-tun

setenv FRIENDLY_NAME "AntiZapret VPN TCP"

**Reply by egorovna-developments** [reply]
Подскажите, как убрать все предустановленные адреса? Youtube и остальное

**Reply by xtrime-ru** [reply]
> Подскажите, как убрать все предустановленные адреса? Youtube и остальное

```shell
docker compose exec -it antizapret-vpn bash
echo '' > /root/antizapret/config/include-hosts-dist.txt
doall
```

**Reply by egorovna-developments** [reply, marked as answer]
Перебрав все настройки понял что upnp для xbox ломает live.com включённый в список vpn. Как только его исключаешь, сразу всё работает как часы.

**Reply by egorovna-developments** [reply]
Причём внесение в exclude-hosts-custom.txt недостаточно. нужно прямо из контейнера include-hosts-dist.txt удалять. Зачем вообще этот домен был добавлен?

**Reply by xtrime-ru** [reply]
microsoft блокирует ru ip на некоторых своих доменах. 
dist не стоит править, надежнее в exclude-regex-custom.txt добавить. Там точно сработает.
